added travel folder and airports

ZionMiller opened the pull request:
### Summary

I often find myself creating these seeds and would love the addition to Faker to help save time on travel apps. This pull contains US and EU airports by size and IATA code, allowing the community to generate random airports to create tickets etc for travel apps. 

### Other Information

        def name(size, region)
          fetch("airport.#{region}.#{size}")
        rescue I18n::MissingTranslationData
          p 'valid arguments are size && region -> US has size large medium small, EU has size large medium -- united_states || european_union'
        end

Produces random US Airport by name.

@return [String]

@example
Faker::Travel::Airport.name('large', 'united_states') => "Los Angeles International Airport"

@faker.version next

stefannibrasil reviewed (commented):
Thank you for your contribution!

The PR body is a being previewed weirdly. Would you mind editing it to be a paragraph?

Could you please remove the comment in the test file? It's not really adding anything :thinking:

ZionMiller commented on test/faker/travel/test_faker_airports.rb:
```diff
@@ -0,0 +1,51 @@
+    @tester = Faker::Travel::Airport
+  end
+
+  # test using assert_match not assert
```
thanks for catching that!

thdaraujo commented on lib/faker/travel/airport.rb:
```diff
@@ -0,0 +1,139 @@
+        # @return [String]
+        #
+        # @example
+        # Faker::Travel::Airport.large_airport_us #=> "Los Angeles International Airport"
```
there is some missing indentation on the documentation:
```suggestion
        #   Faker::Travel::Airport.large_airport_us #=> "Los Angeles International Airport"
```

thdaraujo commented on lib/faker/travel/airport.rb:
```diff
@@ -0,0 +1,139 @@
+        # @example
+        # Faker::Travel::Airport.airport_name #=> "Munich Airport"
+        #
+        # @faker.version next
+
```
extra line:
```suggestion
        # @faker.version next
```

thdaraujo commented on lib/faker/travel/airport.rb:
```diff
@@ -0,0 +1,139 @@
+
+        def small_airport_us
+          fetch('airport.small_airport_us')
+        end
+        ##
```
```suggestion
        end
        
        ##
```

thdaraujo reviewed (changes requested):
thanks for adding this!

There are some issues with the documentation and I left a few suggestions.

thdaraujo commented:
what do you think of passing the continent/region as an argument to the generator?

the reason is that if we decide to add more regions and airports from countries outside of US and Europe in the future, we would have to add new generators as well, probably one for each region.

stefannibrasil commented:
> what do you think of passing the continent/region as an argument to the generator?
> 
> the reason is that if we decide to add more regions and airports from countries outside of US and Europe in the future, we would have to add new generators as well, probably one for each region.
> 
> @ZionMiller @stefannibrasil

I like the idea!

ZionMiller commented:
> > what do you think of passing the continent/region as an argument to the generator?
> > the reason is that if we decide to add more regions and airports from countries outside of US and Europe in the future, we would have to add new generators as well, probably one for each region.
> > @ZionMiller @stefannibrasil
> 
> I like the idea!

Great idea!

stefannibrasil reviewed (commented):
Looking great!

What @thdaraujo was suggesting it to pass the continent/region as an argument.

I think we could even differentiate between generating the name or the IATA code:

```
Faker::Travel::Airport.name(size: 'small', region: 'north_america') => "Palm Springs International Airport"

Faker::Travel::Airport.iata(size: 'large', region: 'europe') => "FRA"
```

Also, could you add the new generator to the [README](https://github.com/faker-ruby/faker/#default)? That way, new users can find this new generator and know how to use it 🌟

ZionMiller commented:
> Looking great!
> 
> What @thdaraujo was suggesting it to pass the continent/region as an argument.
> 
> I think we could even differentiate between generating the name or the IATA code:
> 
> ```
> Faker::Travel::Airport.name(size: 'small', region: 'north_america') => "Palm Springs International Airport"
> 
> Faker::Travel::Airport.iata(size: 'large', region: 'europe') => "FRA"
> ```
> 
> Also, could you add the new generator to the [README](https://github.com/faker-ruby/faker/#default)? That way, new users can find this new generator and know how to use it 🌟

Added that in and updated README 🤝

stefannibrasil reviewed (approved):
Awesome. Thank you for your contribution to faker!

thdaraujo commented:
thanks for the contribution, @ZionMiller!

I'm a bit late, but going back to the original suggestion from @stefannibrasil [here](https://github.com/faker-ruby/faker/pull/2601#pullrequestreview-1169242041), I'm wondering if it would be better for this generator to have keyword arguments instead of positional ones. What do you think?

The reason is that we just moved away from positional arguments on faker v3. I think it would make sense to have it as a standard for new generators too.

cc: @stefannibrasil @ZionMiller

stefannibrasil commented:
ohhh, yes! We totally missed this detail.

I'm going to revert this PR. We don't support positional arguments anymore since v3. @ZionMiller could you make that change and update the docs with the keywords arguments? Thank you!

stefannibrasil commented:
Could you open a new PR wit this new change? Sorry for the confusion, I didn't catch that during the review. Let me know if you have any questions @ZionMiller Thank you!
